### PR TITLE
Use smock to test L2 contracts

### PR DESCRIPTION
**Pull request by i-stam**

### CODE REVIEW

File: contracts/test-helpers/FakeSynthetixBridgeToBase.sol [hunk 1]
```diff
@@ -17,7 +17,7 @@ contract FakeSynthetixBridgeToBase is SynthetixBridgeToBase {
     ) public SynthetixBridgeToBase(_owner, _resolver) {
         mintableSynthetix = ISynthetix(_mockMintableSynthetix);
         xChainBridge = _bridge;
-        crossDomainMessengerMock = address(new MockCrossDomainMessenger(_bridge));
+        crossDomainMessengerMock = address(new MockCrossDomainMessenger());
```
Comment (justinjmoses) [inline contracts/test-helpers/FakeSynthetixBridgeToBase.sol:20]: We can delete this file now no?

File: contracts/test-helpers/MockCrossDomainMessenger.sol [hunk 2]
```diff
@@ -5,16 +5,10 @@ import "../interfaces/ISynthetixBridgeToBase.sol";
 
 
 contract MockCrossDomainMessenger {
-    address public xDomainMsgSender;
```
Comment (justinjmoses) [inline contracts/test-helpers/MockCrossDomainMessenger.sol:8]: We should be able to delete this as well no?

File: test/contracts/SynthetixBridgeToBase.unit.js [hunk 3]
```diff
@@ -25,152 +22,161 @@ contract('SynthetixBridgeToBase (unit tests)', accounts => {
...
+			artifacts.require('SynthetixBridgeToOptimism').abi.find(({ name }) => name === fnc),
+			args
+		);
+
+	describe('when all the deps are (s)mocked', () => {
+		let messenger;
+		let mintableSynthetix;
+		let resolver;
+		beforeEach(async () => {
+			messenger = await smockit(artifacts.require('MockCrossDomainMessenger').abi, {
```
Comment (justinjmoses) [inline test/contracts/SynthetixBridgeToBase.unit.js:36]: Instead of mocking this target, you can mock the interface and it should work as expected

**Review by justinjmoses — APPROVED** [review]
Looking good - just please remove those unused testing files. Otherwise LGTM.

File: contracts/test-helpers/FakeSynthetixBridgeToBase.sol (hunk 1 above)
Comment (i-stam) [reply contracts/test-helpers/FakeSynthetixBridgeToBase.sol:20]: yes, we don't need all those "fake" contracts anymore

**Review by justinjmoses — APPROVED** [review]

**Pull request merged**